src/ResourceManagement/Sql/SqlServerDnsAliasImpl.cs

```diff
@@ -0,0 +1,214 @@
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses DNS alias operations.</param>
+        ///GENMHASH:6EB3A0B36FAECA791EFBB40C3F76F7E2:FC8025F8B79091C30C130C9AAECEC55E
+        internal SqlServerDnsAliasImpl(string name, SqlServerImpl parent, ServerDnsAliasInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, parent)
+        {
+            this.name = name;
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = parent.ResourceGroupName;
+            this.sqlServerName = parent.Name;
+        }
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name.</param>
+        /// <param name="sqlServerName">The parent SQL server name.</param>
+        /// <param name="name">The name of this external child resource.</param>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses DNS alias operations.</param>
+        ///GENMHASH:F6E88B09AC6260EB63A570307518001D:38F5303BD24DCCFCF927664812C8523C
+        internal SqlServerDnsAliasImpl(string resourceGroupName, string sqlServerName, string name, ServerDnsAliasInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, null)
+        {
+            this.name = name;
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = resourceGroupName;
+            this.sqlServerName = sqlServerName;
+        }
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="name">The name of this external child resource.</param>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses DNS alias operations.</param>
+        ///GENMHASH:288BB2F1178A7F50CBD9A307E67BDFB9:394B41296E95933E362FA2C6FB5E1428
+        internal SqlServerDnsAliasImpl(string name, ServerDnsAliasInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, null)
+        {
+            this.name = name;
+            this.sqlServerManager = sqlServerManager;
+            if (innerObject != null && innerObject.Id != null)
+            {
+                try
+                {
+                    ResourceId resourceId = ResourceId.FromString(innerObject.Id);
+                    this.resourceGroupName = resourceId.ResourceGroupName;
+                    this.sqlServerName = resourceId.Parent.Name;
+                } catch (Exception) { }
```

Comment:
> } catch (Exception) { } [](start = 15, length = 24)

please remove catch all.

Reply:
same as the previous case, inner.Id might not be available if the inner is constructed rather than retrieved from the service